### DISCUSSION TITLE
Directory structure

### DISCUSSION THREAD

**Discussion by Leon0402** [friendlyanon/cmake-init#19] [Q&A]
Hi, 

I wonder what the choice was for the directory names. Does this follow a standard? 

Especially `source` is something I never saw before, I only know it as `src`. I personally would also prefer `tests` instead of `test`

**Comment by friendlyanon** [comment]
Noone can follow a standard in this regard, because there is no standard. It's loosely based on the pitchfork layout.

My preference for `source` instead of `src` is merely due to my personal taste. Same goes for `test`. They are trivial to change regardless.

A CLI option to customize this could be added I guess, but I don't see the value, since like I said, there is no standard.

**Reply by Leon0402** [reply]
Hmm yeah I assumed there isn't really a standard. Especially not for `test` vs `tests`. For the `source` vs `src`thing I would say `src` is the most common in C++ and programming in general. But that's mostly based on my experience, so could be wrong.
If you feel the same way I would suggest making this the default as most people will be happy with it, in contrast to `source` perhaps. 

But an option might be nice as well. Yes it's not super difficult to change, you're right about it. You need to change a few things though like the include directory path for the lib. If you forget this, you will have errors with your tests. 
I could imagine this to be a "stupid mistake" one can make and search for like half an hour :-) 
In my opinion `src` could deserve it's option. For `test` I agree with you, that's really trivial to change and not very error-prone.

**Reply by friendlyanon** [reply]
In-source builds are probably still very prevalent due to the legacy of makefile based projects, but so is not using CMake right. The decision for the source directory to bear the `src` name is merely historical at this point and I feel it's unnecessarily abbreviated.

I don't know, changing `source` to `src` is very methodical and programmers are good at solving those kinds of things. Doing a rename in a freshly generated project is simple:
```sh
find . -type f -not -path '*/.git/*' -exec sed -i -Ee 's@(\\\\|/)source@\1src@g' -e 's@source/@src/@g' {} \;
```

But then again, what does one gain with using `src` over `source`?

**Reply by Leon0402** [reply]
The gain is to have a somewhat standard, even if it's only for historical reasons. Some are bad for sure, I think this is fine. It makes it a lot easier for me at least if as many project as possible use a similar directory structure. I have nothing against source, I just don't think it's going to change towards source.

But I can live with how it is. As you said find & replace is easy! :-)
